bls/bls.go

```diff
@@ -1,79 +1,110 @@
 package bls
 
 import (
-	"crypto/rand"
 	"errors"
+	"math/big"
 
-	bls12381 "github.com/kilic/bls12-381"
+	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	"github.com/consensys/gnark-crypto/ecc/bls12-381/fr"
 )
 
-// Heavily inspired by:
-// https://github.com/protolambda/bls12-381-util/blob/master/signatures.go
-// Thank you for the excellent code.
-
-var domain = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
-
 const (
-	PublicKeyLength int = 48
-	SecretKeyLength int = 32
-	SignatureLength int = 96
+	PublicKeyLength = bls12381.SizeOfG1AffineCompressed
+	SecretKeyLength = fr.Bytes
+	SignatureLength = bls12381.SizeOfG2AffineCompressed
 )
 
 type (
-	PublicKey = bls12381.PointG1
```

Comment:
These conversions are correct. Its probably better to store these as Affine like how you did it and then if you need to do ScalarMultiplication, we can convert them to Jacobian :)